Plugins/ThreadWatchdog/ThreadWatchdog.cpp

```diff
@@ -24,12 +25,18 @@ static Hooks::Hook s_MainLoopHook;
 ThreadWatchdog::ThreadWatchdog(Services::ProxyServiceList* services)
     : Plugin(services)
 {
-    s_MainLoopHook = Hooks::HookFunction(&CServerExoAppInternal::MainLoop,
-                                                  &MainLoopUpdate, Hooks::Order::Earliest);
-
     s_watchdogPeriod = Config::Get<uint32_t>("PERIOD", 15);
     // Default to effectively infinite
     s_watchdogKillThreshold = Config::Get<uint32_t>("KILL_THRESHOLD", ~0);
+
+    static Hooks::Hook loadModuleStartHook = Hooks::HookFunction(&CNWSModule::LoadModuleStart,
```

Comment:
I have no clue why this is needed. The commit message is very light on details. Such a hack, if really needed, needs a detailed comment in the code and commit message.